types/fusesoc-definition.json

```
@@ -0,0 +1,33 @@
+{
+  "$schema": "https://packageurl.org/schemas/purl-type-definition.schema-1.0.json",
+  "$id": "https://packageurl.org/types/fusesoc-definition.json",
+  "type": "fusesoc",
+  "type_name": "FuseSoC core",
+  "description": "FuseSoC core",
+  "repository": {
+    "use_repository": true,
+    "default_repository_url": "https://cores.fusesoc.net"
+  },
+  "namespace_definition": {
+    "native_name": "vendor/library",
+    "case_sensitive": true,
+    "note": "The namespace contains both the vendor and library part of the fusesoc core name.",
+    "requirement": "required"
+  },
+  "name_definition": {
+    "requirement": "required",
+    "native_name": "name",
+    "case_sensitive": true
+  },
+  "version_definition": {
+    "requirement": "optional",
+    "native_name": "version",
+    "note": "The version is the semantic version of the fusesoc core.  We would really like this to be required, but that breaks validation."
```

I am not sure you need this comment.... The fact that version is optional is a spec thing when you want to refer to a package without mentioning a version like in the CVE schema, or when using that with a VERS. What about this:

```suggestion
    "note": "The version is the semantic version of the fusesoc core."
```